core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -200,7 +210,18 @@ private[spark] class AppClient(
 
       case r: RequestExecutors =>
         master match {
-          case Some(m) => context.reply(m.askWithRetry[Boolean](r))
+          case Some(m) =>
+            receiveAndReplyThreadPool.execute(new Runnable {
```

Comment:
At first I was trying to put these 2 calls into something like `Utils.tryWithSafeFinally` with blocks  as arguments but the call ended up looking a little confusing as to what was happening.

maybe just a regular function like this would be better, although a little less flexible

```scala
private def receiveAndReplyAsync[T](masterRef: RpcEndpointRef, context: RpcCallContext,
                                     msg: T): Unit = {
  // execute ask and reply in thread pool
  ..
```